### PR TITLE
fix: Fix issues and implement usbl_pose port

### PR DESCRIPTION
**Pull request by DeboraTahara**
Depends on:
- [x] https://github.com/tidewise/drivers-usbl_seatrac/pull/6
- [x] https://github.com/rock-core/drivers-orogen-iodrivers_base/pull/32
- [ ] https://github.com/tidewise/bundles-seabots/pull/153

### CODE REVIEW

File: tasks/Task.cpp [hunk 1]
```diff
@@ -86,8 +86,9 @@ void Task::updateHook()
     _ping_status.write(ping);
     if (ping.flag == 1) {
         auto rbs = convertToRBS(ping);
-        _local_pose.write(rbs);
+        _pose.write(rbs);
         auto attitude = convertToQuaterniond(ping);
+        _local_attitude.write(attitude);
```
Comment (jhonasiv) [inline tasks/Task.cpp]: Why not just pass the orientation to the local pose? It actually describes the local pose frame

File: tasks/Task.cpp [hunk 2]
```diff
@@ -20,11 +20,11 @@ Task::~Task()
...
     Eigen::Quaterniond attitude = Eigen::Quaterniond(
-        Eigen::AngleAxisd(data.response.acoustic_fix.attitude_roll / 10. / 180.0 * M_PI,
+        Eigen::AngleAxisd(data.response.acoustic_fix.attitude_yaw / 10. / 180.0 * M_PI,
             Eigen::Vector3d::UnitX()) *
         Eigen::AngleAxisd(data.response.acoustic_fix.attitude_pitch / 10. / 180.0 * M_PI,
             Eigen::Vector3d::UnitY()) *
-        Eigen::AngleAxisd(data.response.acoustic_fix.attitude_yaw / 10. / 180.0 * M_PI,
+        Eigen::AngleAxisd(data.response.acoustic_fix.attitude_roll / 10. / 180.0 * M_PI,
             Eigen::Vector3d::UnitZ()));
     return attitude;
```
Comment (jhonasiv) [inline tasks/Task.cpp]: This is wrong, yaw should be on Z axis and roll on X (you did convertToRBS right, just check it out) @kapeps  mentioned that you are not going to use this anymore, but I thought you should know this error was here anyway.

**Review by kapeps — APPROVED** [review]

File: usbl_seatrac.orogen [hunk 3]
```diff
@@ -22,10 +22,47 @@ task_context "Task", subclasses: "iodrivers_base::Task" do
...
+    property "xcvr_posflt_tmo", "base/Time"
+    # How often periodic status output messages are generated
+    property "status_mode",  "usbl_seatrac/protocol/StatusMode",
+             "STATUS_MODE_2HZ5"
 
-    output_port("pose", "base/samples/RigidBodyState")
+    # USBL Local Relative Position Coordinates
+    output_port("local_pose", "base/samples/RigidBodyState")
+    # USBL Beacon X150  - z = local_pressure , orientation = local_attitude
+    output_port("usbl_pose", "base/samples/RigidBodyState")
```
Comment (jhonasiv) [inline usbl_seatrac.orogen]: I really dislike usbl_pose as the port name, maybe change it to `transceiver_pose` or something like that?

File: tasks/Task.cpp [hunk 4]
```diff
@@ -16,17 +16,69 @@ Task::~Task()
...
 
     rbs.time = data.timestamp;
-    rbs.position = Eigen::Vector3d(data.response.acoustic_fix.position.north,
-        -data.response.acoustic_fix.position.east,
-        -data.response.acoustic_fix.position.depth);
+    rbs.position = Eigen::Vector3d(data.response.acoustic_fix.position.north / 10.0,
+        data.response.acoustic_fix.position.east / 10.0,
+        data.response.acoustic_fix.position.depth / 10.0);
     return rbs;
 }
```
Comment (jhonasiv) [inline tasks/Task.cpp]: I think naming both functions the same is confusing, if we come back to this in a couple of months we would curse our previous selves. Just name the first one to convertToZWithOrientationRBS or something like that and the second to convertToPositionRBS

File: tasks/Task.cpp [hunk 5]
```diff
@@ -16,17 +16,69 @@ Task::~Task()
 {
 }
 
-static RigidBodyState convertToRBS(PingStatus const& data)
+static RigidBodyState convertToZWithOrientationRBS(Status const& data)
+{
+    RigidBodyState rbs;
+    rbs.time = data.timestamp;
+    rbs.position = Eigen::Vector3d(0, 0, data.environment.depth / 10.);
```
Comment (jhonasiv) [inline tasks/Task.cpp]: This is used to report a Pose, (0,0) is a valid coordinate but in this case we dont have any valid data for X and Y. You should be passing a Vector3d as `(NaN, NaN, data.environment.depth / 10.0)`

File: usbl_seatrac.orogen [hunk 6]
```diff
@@ -22,10 +22,47 @@ task_context "Task", subclasses: "iodrivers_base::Task" do
...
+    property "xcvr_resp_time", "base/Time"
+    # Maximum time that a beacon is not communicated with before its position filter
+    # is reset
+    property "xcvr_posflt_tmo", "base/Time"
+    # How often periodic status output messages are generated
+    property "status_mode",  "usbl_seatrac/protocol/StatusMode",
+             "STATUS_MODE_2HZ5"
 
-    output_port("pose", "base/samples/RigidBodyState")
+    # USBL Local Relative Position Coordinates
```
Comment (doudou) [inline usbl_seatrac.orogen]: The USBL only returns position. The port should be remote2local_position and the comment should be "Position of the remote transponder in relation to the master's device body pose"

File: usbl_seatrac.orogen [hunk 7]
```diff
@@ -22,10 +22,47 @@ task_context "Task", subclasses: "iodrivers_base::Task" do
...
+    output_port("remote2local_pose", "base/samples/RigidBodyState")
+    # USBL Beacon X150  - z = local_depth , orientation = local_attitude
+    output_port("transceiver_pose", "base/samples/RigidBodyState")
```
Comment (doudou) [inline usbl_seatrac.orogen]: local2nwu_orientation_with_z

**Review by doudou — APPROVED** [review]

**Pull request merged**